module/zfs/dsl_dataset.c

```diff
@@ -73,12 +73,12 @@
  * The SPA supports block sizes up to 16MB.  However, very large blocks
  * can have an impact on i/o latency (e.g. tying up a spinning disk for
  * ~300ms), and also potentially on the memory allocator.  Therefore,
- * we do not allow the recordsize to be set larger than zfs_max_recordsize
- * (default 1MB).  Larger blocks can be created by changing this tunable,
- * and pools with larger blocks can always be imported and used, regardless
- * of this setting.
+ * we did not allow the recordsize to be set larger than zfs_max_recordsize
+ * (former default: 1MB).  Larger blocks could be created by changing this
+ * tunable, and pools with larger blocks could always be imported and used,
+ * regardless of this setting.
  */
-int zfs_max_recordsize = 1 * 1024 * 1024;
+int zfs_max_recordsize = 16 * 1024 * 1024;
```

Comment:
Would it be better to keep it at 1MB by default on 32bit platforms instead of setting RECORDSIZE_PERM to RO at the bottom of the file, that makes no any sense after this.